Lecture3/Task1/product/ProductFactoryDirector.java

```diff
@@ -0,0 +1,44 @@
+package product;
+
+import builder.AcousticGuitarBuilder;
+import builder.ElectricGuitarBuilder;
+import builder.Guitar;
+import builder.GuitarBuilderDirector;
+import exceptions.FailedOrderException;
+import utils.RandomUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ProductFactoryDirector {
+
+	public static List<Guitar> createGuitars() throws FailedOrderException {
+		List<Guitar> guitars = new ArrayList<>();
+		GuitarBuilderDirector builderDirector = new GuitarBuilderDirector();
+		if (guitars.size() == 0) {
+			int amount = RandomUtils.getAmount();
```

Comment:
you have to provide max amount from client side.  RandomUtils.getAmount(int maxAmount);
as it is hard to understand what amount this method return in your case